flink-libraries/flink-table/src/test/scala/org/apache/flink/table/formats/utils/TestAmbiguousTableFormatFactory.scala

```diff
@@ -16,22 +16,32 @@
  * limitations under the License.
  */
 
-package org.apache.flink.streaming.connectors.kafka;
+package org.apache.flink.table.formats.utils
 
-import static org.apache.flink.table.descriptors.KafkaValidator.CONNECTOR_VERSION_VALUE_010;
+import java.util
+
+import org.apache.flink.table.formats.TableFormatFactory
+import org.apache.flink.types.Row
 
 /**
- * Tests for {@link Kafka010AvroTableSourceFactory}.
- */
-public class Kafka010AvroTableSourceFactoryTest extends KafkaAvroTableSourceFactoryTestBase {
+  * Table format factory for testing.
+  */
+class TestAmbiguousTableFormatFactory extends TableFormatFactory[Row] {
```

Comment:
shouldn't this extend or reuse `TestTableFormatFactory`? They are now interconnected implicitly - you assume some dependencies between those two classes - and this can veerryy easily brake (silently) if someone modifies only one of those classes forgetting about the other. (silently - tests might still pass while they will not be testing what you intended anymore).

Reply:
This implicit interconnection is on purposes. A similar configuration could happen in a SQL Client library folder. The `TableFormatFactoryServiceTest` should fail if they are changed too much. But I added more comments for implementers. If `TestAmbiguousTableFormatFactory` would extend from `TestTableFormatFactory` then the class based selection would not work anymore.